security.md

```diff
@@ -0,0 +1,97 @@
+machine-in-the-middle (MitM) attack. For this, they quickly reconnect with
+the same code in order to connect with the second peer. With that, they get a
+secure channel to each side while being able to read and modify the connection's
+content.
+
+Both cases can be mitigated by enabling the verifier feature, in which case
+even successful attacks can be detected.
+
+In general, this is a very similar process to ordinary key exchanges like
+Diffie-Hellman (or variations), except that MitM attackers additionally have to
+guess the code correctly. (Hence the name, "*password authenticated* key
+exchange", PAKE.)
+
+### Password entropy
+
+Compared to password strengths we are used to nowadays, 16 bits may seem scarily
+small, even knowing that an attacker only has one try to guess the password. But
+to put it into perspective: As a user, you'd have to experience 32000 failed
+transfers because of an attack in order to having a 50% chance of a file being
+intercepted. From an attacker's perspective, guessing thousands of codes is not
+difficult, but there is no possibility of targetting any individual connection,
+making any gained data fairly useless. From a service perspective, any successful
+attack implies 65536 failed attempts on average, which will quickly be noticed
+due to their disrupting nature.
+
+In short, guessing codes is unlikely to be an attractive target for any attacker.
+Of course, there is always the possibility to configure a longer code in the
+clients if desired.
+
+## Denial of Service (DoS) attacks on the Mailbox server
+
+Wormhole codes can be so short because they implicitly contain a common
+rendezvous server URL (any two applications that use Magic Wormhole
+should be configured to use the same server). As a result, successful
+operation depends upon both clients being able to contact that server,
+making it a single point of failure (SPOF).
+
+While it is of course possible to incapacitate the rendezvous server using
+traditional means of DoS, so much work is not necessary: An attacker may
+simply connect to every nameplate with a random code, causing the key exchange
+to fail. There is a "list" command in the protocol which makes it easy to
+enumerate all nameplates to disrupt them, but even without it the name space is
+sufficiently small (by design, as we want short codes) to brute force.
+
+There are several ideas to defend against this, however the only one already
+specified in the protocol is the "permission" feature: The rendezvous server
+may use it to restrict new connections with some kind of challenge (or for
+private servers, authentication). The motivation behind this is to enable
+a proof-of-work challenge (using HashCash) when under attack to slow it down.
+See the server protocol specification for details.
+
+Currently the best defense against this is to use high-entropy nameplates which
+cannot easily be enumerated at the cost of longer codes. This has not be
```

Comment:
If we're going to leave out speculation, maybe this too? (I'd have to do a lot more thinking to convince myself this is "the best" for example). Also typo: "This has not be" -> "This has not been"

Reply:
Yes, we can leave it out for now. But actually I have some old proposal for "high entropy nameplates" lying around, I could dig it out again and do this, should not be too hard in general.

Reply:
If users can exchange high-entropy nameplates, can't they just exchange a key directly?
(I guess I'm assuming "high entropy" here means "32 bytes" ...)